### PR TITLE
Add option to automatically turn off if the 'list' option is off

### PR DESCRIPTION
**Pull request by zeertzjq**
This allows toggling two thing with one command, or one mapping (`yol` from `vim-unimpaired`), and reduces the need to keep a list of disabled filetypes because many filetypes (e.g. `help`) have `ftplugin`s that unset `list`.

### CODE REVIEW

File: lua/indent_blankline/utils.lua [hunk 1]
```diff
@@ -68,6 +69,9 @@ M.is_indent_blankline_enabled =
         if g_enabled ~= true then
             return false
         end
+        if respect_list then
+          return vim.opt.list:get()
```
Comment (lukas-reineke) [inline lua/indent_blankline/utils.lua:73]: This should not overwrite the other checks.
if it is `false`, return `false`, if it is `true`, continue with the other checks.

And this function is memoized, you have to get the value in `init.lua` and pass it as an argument

File: plugin/indent_blankline.vim [hunk 2]
```diff
@@ -31,6 +31,7 @@ lua require("indent_blankline").init()
 
 augroup IndentBlanklineAutogroup
     autocmd!
+    autocmd OptionSet list lua require("indent_blankline.utils").list_changed()
```
Comment (lukas-reineke) [inline plugin/indent_blankline.vim:34]: you don't need this `list_changed` function, just  add it to the other `OptionSet` autocommand. `IndentBlanklineRefresh` will do the right thing

**Review by lukas-reineke — CHANGES_REQUESTED** [review]
Thanks, I like it
maybe it should even be on by default

**Comment by zeertzjq** [comment]
> This allows toggling two thing with one command, or one mapping (`yol` from `vim-unimpaired`), and reduces the need to keep a list of disabled filetypes because many filetypes (e.g. `help`) have `ftplugin`s that unset `list`.

For users who don't have `'list'` set having this on by default changes the behavior.

File: lua/indent_blankline/init.lua [hunk 3]
```diff
@@ -91,6 +92,8 @@ local refresh = function()
         not utils.is_indent_blankline_enabled(
             vim.b.indent_blankline_enabled,
             vim.g.indent_blankline_enabled,
+            vim.g.indent_blankline_respect_list,
```
Comment (lukas-reineke) [inline lua/indent_blankline/init.lua:95]:
```suggestion
            v("indent_blankline_respect_list"),
```

**Review by lukas-reineke — CHANGES_REQUESTED** [review]
right, `list` is not on by default. Then let's go with this off by default as well :+1: 

Code looks good now, just one last change

**Review by lukas-reineke — APPROVED** [review]
Thank you!

**Pull request merged**